translate/translate.go

```diff
@@ -129,6 +134,85 @@ type Key struct {
 
 type informationElements map[Key]InformationElementEntry
 
+var reducedSizeErr error = errors.New("Unable to read reduced size encoding")
+
+// Helper method to read an unsigned reduced size field
+func reducedSizeReadUnsigned(bs []byte, maxBits int) (uint64, error) {
+	// Exit if `bs` has more bits than we can store
+	if len(bs)*8 > maxBits {
+		return 0, reducedSizeErr
```

Comment:
nit: In the 3 situations where there are too many bits, maybe return a different, more specific error like "Unable to read reduced size encoding because it is too big"